src/StoragelessSessionPersistence2.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Chubbyphp\Session\Storageless;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use PSR7Sessions\Storageless\Http\SessionMiddleware as PSR7SessionMiddleware;
+use Zend\Expressive\Session\SessionPersistenceInterface;
+use Zend\Expressive\Session\SessionInterface;
+use PSR7Session\Session\SessionInterface as PSR7SessionInterface;
+use Zend\Expressive\Session\Session;
+
+final class StoragelessSessionPersistence2 implements SessionPersistenceInterface
+{
+    /**
+     * @var ServerRequestInterface
+     */
+    private $request;
+
+    /**
+     * Generate a session data instance based on the request.
+     */
+    public function initializeSessionFromRequest(ServerRequestInterface $request): SessionInterface
+    {
+        $this->request = $request;
+
+        /** @var PSR7SessionInterface $psr7StoragelessSession */
+        $psr7StoragelessSession = $this->request->getAttribute(PSR7SessionMiddleware::SESSION_ATTRIBUTE);
+        $session = new Session((array) $psr7StoragelessSession->jsonSerialize());
```

Comment:
I think that providing an adapter to the session as well (as I suggested earlier), would remove the need having a stateful component...

The signature of this method accepts **any** implementation of `SessionInterface`. So, if you create an adapter that implements Zend's interface and wraps ours, you'd be able to simplify things by a lot here.

Reply:
Something like:

```php
namespace Chubbyphp\Session\Storageless;

final class Session implements \Zend\Expressive\Session\SessionInterface
{
    private $session;

    public function __construct(\PSR7Session\Session\SessionInterface $session)
    {
        $this->session = $session;
    }

    // Then you forward all interface methods to `$this->session` :) 
}

Reply:
@lcobucci works, thanks for solving the issue. @Ocramius should i finish the work and keep this repo or do you wanna add this to psr7-sessions org, as part of the existing repo or as a new?